Integrations/integration-ATD.yml

```diff
@@ -37,672 +38,670 @@ configuration:
+            files=files
+        )
+        if res.status_code < 200 or res.status_code >= 300:
+                return_error('Got status code ' + str(res.status_code) + ' with body ' + res.content + ' with headers ' + str(res.headers))
+        result = res.content
+        if not uri.startswith('php/showreport.php?'):
+            result = json.loads(result)
+            if 'success' in result:
+                if (result['success'] == 'false'):
+                    return_error('ATD Api call to ' + uri + ' failed with request body: ' + body)
+        return result
+
+
+
+    ''' FUNCTIONS '''
+    def heart_beat():
+        return http_request('php/heartbeat.php','get', API_HEADERS, HEARTBEATHEADERS)
+
+    def test_get_session():
+        res = http_request('php/session.php','get', LOGINHEADERS)
+
+    def prettify_current_user_res(currentUser):
+        prettyCurrentUser = {
+            'APIVersion': currentUser['apiVersion'],
+            'IsAdmin': 'True' if (currentUser['isAdmin'] == '1') else 'False',
+            'SessionId': currentUser['session'],
+            'UserId': currentUser['userId']
         }
-        return res;
-    }
-
-    function sendRequest(uri, method, headers, body){
-        body = sendRequestRaw(uri,method, headers, body).Body;
-        var res = JSON.parse(body);
-        if (res.success) {
-            if (res.success === 'false') {
-                throw 'ATD Api call to '+uri+' failed with request body: '+body;
+        return prettyCurrentUser
+
+    def get_session():
+        res = http_request('php/session.php','get', LOGINHEADERS)
+        md = tableToMarkdown('ATD Current User',prettify_current_user_res(res['results']),
+                                ['APIVersion', 'IsAdmin', 'SessionId', 'UserId'])
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res['results'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': md,
+            'EntryContext': {
+                  "ATD.Session(val.SessionId==obj.SessionId)": prettify_current_user_res(res['results'])
             }
+        })
+
+    def prettify_list_users_res(users):
+        prettyUsers = []
+
+        for i in range(len(users)):
+            user = users[i]
+            prettyUsers.append({
+                'FullName': user['fullName'],
+                'UserId': user['idx'],
+                'LoginId': user['loginId'],
+                'UserType': user['userType']
+            })
+
+        return prettyUsers
+
+    def list_users(userType):
+        userType = userType if userType else 'STAND_ALONE'
+        result = http_request('php/briefUserList.php?userType='+userType,'get', API_HEADERS)
+        users = result['results']
+        return users
+
+    def list_users_command():
+        users = list_users(demisto.args()['userType'])
+
+        prettyUsers = prettify_list_users_res(users)
```

Comment:
Please handle the case no results found, by returning a string and not an empty table.

Reply:
done